wasm/src/signature.rs

```diff
@@ -107,10 +49,66 @@ pub async fn authorize_withdrawal(
     let contract_transfer = to_contract_external_transfer(&transfer)
         .map_err(|_| String::from("Failed to convert transfer"))?;
 
-    let buf = postcard::to_allocvec(&contract_transfer)
+    let contract_transfer_bytes = postcard::to_allocvec(&contract_transfer)
         .map_err(|e| format!("Failed to serialize transfer: {e}"))?;
 
-    let digest = keccak256(&buf);
+    sign_message(
+        wallet,
+        seed,
+        &contract_transfer_bytes,
+        key_type,
+        external_signer,
+    )
+    .await
+}
+
+/// Signs a message using either internal or external signing method.
+///
+/// For internal signing, uses the seed to derive a signing key.
+/// For external signing, uses the provided external_signer function.
+pub async fn sign_message(
+    wallet: &Wallet,
+    seed: &str,
+    message: &[u8],
+    key_type: &str,
+    external_signer: Option<&Function>,
```

Comment:
for these methods we could avoid needing to pass `key_type` and instead just check:
```rust
match external_signer {
    Some(callback) => sign_with_external_key(msg, callback),
    None => sign_with_internal_key(wallet, seed, message),
}
```

Not that opinionated though, this works too